JamGame/Assets/Scripts/TileUnion/PlaceCondition.cs

```diff
@@ -30,10 +30,9 @@ public class RequiredTile : IPlaceCondition
         [SerializeField]
         private List<string> requiredTags;
 
-        public Result ApplyCondition(
+        public List<TileImpl> ApplyCondition(
```

Comment:
Better introduce new struct with `bool` and `List` because this is very implicit API